sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_polling.py

```diff
@@ -0,0 +1,82 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+from azure.core.exceptions import HttpResponseError
+from azure.core.polling.base_polling import (
+    LocationPolling,
+    OperationResourcePolling,
+    _is_empty,
+    _as_json,
+    BadResponse
+)
+
+
+class TrainingPolling(LocationPolling):
+    """Polling method overrides for training endpoints.
+    """
+
+    def get_polling_url(self):
+        # type: () -> str
+        """Return the polling URL.
+        """
+        return self._location_url + "?includeKeys=true"
+
+    def get_status(self, pipeline_response):  # pylint: disable=no-self-use
+        # type: (PipelineResponseType) -> str
+        """Process the latest status update retrieved from a 'location' header.
+
+        :param azure.core.pipeline.PipelineResponse response: latest REST call response.
+        :raises: BadResponse if response has no body.
+        """
+        response = pipeline_response.http_response
+        if response.status_code == 200:
+            body = _as_json(response)
+            status = body['modelInfo']['status']
+            if not status:
+                raise BadResponse("No status found in body")
+            if status.lower() != "creating":
+                return "Succeeded"
+
+            return "InProgress"
```

Comment:
Detail: you could return "creating", since a random string is always considered "inprogress"